include/s3select.h

```diff
@@ -2076,9 +2137,13 @@ class base_s3object
         {
           a.second->invalidate_cache_result();
         }
+      }
+      while (multiple_row_processing() && m_where_clause && !m_where_clause->eval().is_true() && !(m_is_limit_on && m_processed_rows == m_limit));
 
+      if(m_where_clause && !m_where_clause->eval().is_true() && m_is_limit_on && m_processed_rows == m_limit)
```

Comment:
calling to `where_clause->eval()` twice may consume much more CPU than required. 
calls to `->eval()` should reduce to the minimum.